src/components/Modal/Modal.unit.spec.jsx

```diff
@@ -17,34 +17,48 @@ describe('<Modal/>', () => {
     tree.unmount();
   });
 
-  it('should setup event listeners and tear them down', () => {
-    sinon.spy(global.document, 'addEventListener');
-    sinon.spy(global.document, 'removeEventListener');
+  describe('event listeners', () => {
+    const checkForListener = (callsArray, listenerName) => {
+      return callsArray.filter(call => {
+        return call.firstArg === listenerName;
+      });
+    };
 
-    const totalListeners = 3;
-    const tree = mount(
-      <Modal
-        id="modal"
-        title="Modal title"
-        visible
-        clickToClose
-        onClose={() => {}}
-      >
-        Modal contents
-      </Modal>,
-    );
+    let tree;
+    before(() => {
+      sinon.spy(global.document, 'addEventListener');
+      sinon.spy(global.document, 'removeEventListener');
+    });
 
-    expect(global.document.addEventListener.callCount).to.be.equal(
-      totalListeners,
-    );
+    after(() => {
+      global.document.addEventListener.restore();
+      global.document.removeEventListener.restore();
+    });
 
-    tree.unmount();
-    expect(global.document.removeEventListener.callCount).to.be.equal(
-      totalListeners,
-    );
+    it('should set up a keydown and a click listener', () => {
```

Comment:
Refactored these tests. I wanted to find a way to test for just the listeners we are adding and not the listeners being added by react-focus-on, but this was as close as I could get.